authorization/permission/service/privilege_cache_service.go

```diff
@@ -67,7 +67,10 @@ func (s *privilegeCacheServiceImpl) CachedPrivileges(ctx context.Context, identi
 				ExpiryTime: time.Now().Local().Add(time.Second * time.Duration(s.conf.GetPrivilegeCacheExpirySeconds())),
 			}
 
-			err = s.Repositories().PrivilegeCacheRepository().Create(ctx, privilegeCache)
+			err = s.ExecuteInTransaction(func() error {
```

Comment:
what about having the whole method body in a single transaction ?

Reply:
also, what about making the logic simpler, using the GORM `save` method to either insert or update the record, after the change you made in https://github.com/fabric8-services/fabric8-auth/pull/637/files#diff-55ea2c2abd467421bda2d07c8642a7f7R139?

Reply:
The transaction is only required if there's a DML operation required.  Gorm's `Create()` operation is only used to create a new record, while `Save()` is used to update all fields of a record.